Userland/Libraries/LibWeb/HTML/FormControlInfrastructure.cpp

```diff
@@ -40,4 +46,140 @@ WebIDL::ExceptionOr<Entry> create_entry(JS::Realm& realm, DeprecatedString const
     };
 }
 
+// https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#constructing-the-form-data-set
+WebIDL::ExceptionOr<Optional<HashMap<DeprecatedString, Vector<Variant<JS::Handle<FileAPI::File>, DeprecatedString>>>>> construct_entry_list(JS::Realm& realm, HTMLFormElement& form)
```

Comment:
Oh wow that's a lot of greater than signs :wideeyes: Feel free to add a type definition to make things like this easier.